packages/libs/eda/src/lib/map/analysis/MapAnalysis.tsx

```diff
@@ -123,8 +121,11 @@ export function MapAnalysis(props: Props) {
   );
   const geoConfigs = useGeoConfig(useStudyEntities());
   const location = useLocation();
-  const locationState = location.state as LegacyRedirectState;
-  const [showRedirectModal, setShowRedirectModal] = useState(!!locationState);
+  const locationState = location.state;
+  const [showRedirectModal, setShowRedirectModal] = useState(
+    LegacyRedirectState.is(locationState) &&
+      locationState?.showLegacyMapRedirectModal
```

Comment:
Doing this (or `Boolean(...)`) would keep this state variable 'pure' Boolean which would remove the need to cast it below.

```suggestion
      !!locationState?.showLegacyMapRedirectModal
```